File: MMF_main/Stepper.cpp
#include "Arduino.h"
#include "Stepper.h"

int dirPin;
int stepPin;
int stepsPerUnit; // Steps per unit

void Stepper::init_stepper(int dir, int step, int stepsPU)
{
  dirPin = dir;
  stepPin = step;
  stepsPerUnit = stepsPU;

  pinMode(stepPin, OUTPUT);
  pinMode(dirPin, OUTPUT);
}

void Stepper::move_stepper(float dist_unit, float vel_UpS, int dir)
{
  //dist = 2, vel = 1;

  //time = vel/dist
  int total_steps = dist_unit * stepsPerUnit; //400
  int steps_psec = vel_UpS * stepsPerUnit; // 200
  int time  = total_steps/steps_psec; // 2
  //int steps_psec = total_steps/time;
  double sec_pstep = float(time)/total_steps; // 0,005
  int intsec_pstep = (sec_pstep * 1000.0);

  for(int i=total_steps; i>0; i--)
  {
    step_stepper(dir);
    delay(intsec_pstep);
    //delay(5);
  }
}

void Stepper::step_stepper(int dir)
{
  if(dir == 0) //Clockwise
  {
    digitalWrite(dirPin, HIGH);
  }
  else
  {
    digitalWrite(dirPin, LOW);
  }
  digitalWrite(stepPin, HIGH);
  digitalWrite(stepPin, LOW);
}

void Stepper::set_steps_unit(int spu)
{
  stepsPerUnit = spu;
}

File: MMF_main/Stepper.py
import utime
import time
import machine

class Motor:
    dir_pin = ""
    step_pin = ""
    steps_unit = ""
    def __init__(self, d, s, u):
        global dir_pin
        global step_pin
        global steps_unit
        self.d = d
        self.s = s
        self.u = u
        
        step_pin = machine.Pin(s, machine.Pin.OUT)
        dir_pin = machine.Pin(d, machine.Pin.OUT)
        steps_unit = u
        
        step_pin.value(0)
        dir_pin.value(0)
        print("Init Motor Class")
        
    def step_stepper(self, direction):        
        dir_pin.value(direction)
        step_pin.value(0)
        step_pin.value(1)
   
    def move_motor(self, dist_unit, vel_UpS, direction):
        #print("###")
        
        total_steps = dist_unit * steps_unit
        steps_psec = vel_UpS * steps_unit
        tim = total_steps/steps_psec
        #
        sec_pstep = tim/total_steps
        
        ###
        for i in range(total_steps):
            self.step_stepper(direction)
            time.sleep(sec_pstep)